crates/infra/cli/src/commands/publish/mod.rs

```diff
@@ -16,6 +16,31 @@ use crate::utils::ClapExtensions;
 #[derive(Clone, Debug, Parser)]
 pub struct PublishController {
     command: PublishCommand,
+
+    #[arg(long)]
+    dry_run: bool,
+}
+
+#[derive(Clone, Copy)]
+enum DryRun {
```

Comment:
nit: given that there is no callsite to pass the boolean literal here (CLI), I wonder what is the benefit of the new `DryRun` enum? I think a simple `dry_run: bool` here would be more readable over the extra `DryRun::is_yes()` abstraction.

Reply:
It is definitely used in the code and passed to callers, see `PublishController::execute`.

If I used a bare boolean, then we'd have functions reading `publish_xyz(true/false)`, which would be confusing, especially since first intuition for `publish_xyz(true)`seems to be "yes, really *do* publish it" rather than "yes, this is just a dry run without actually publishing".

I'm fine with not including the dry_run at all. I was paranoid and I wanted a way to explicitly do a dry run in CI, even if we did not provide tokens in my testing CI run.

Reply:
> I was paranoid and I wanted a way to explicitly do a dry run in CI, even if we did not provide tokens in my testing CI run.

Sounds good.

Reply:
You mentioned it is useful to still be able to do a dry run in CI, so I'm fine with keeping it as-is for now.